lib/active_model/serializer/adapter.rb

```diff
@@ -32,8 +35,36 @@ def self.adapter_class(adapter)
         "ActiveModel::Serializer::Adapter::#{adapter.to_s.classify}".safe_constantize
       end
 
+      def fragment_cache(*args)
+        raise NotImplementedError, 'This is an abstract method. Should be implemented at the concrete adapter.'
+      end
+
       private
 
+      def cache_check
```

Comment:
Updated the old ```cache_object``` to ```cache_check```.
Also abstracted some of its behaviours into other methods. (bellow)